packages/native/src/useLinking.native.tsx

```diff
@@ -58,8 +58,9 @@ export default function useLinking(
 
   const extractPathFromURL = React.useCallback((url: string) => {
     for (const prefix of prefixesRef.current) {
-      if (checkIsPrefixMatch(url, prefix)) {
-        return url.replace(prefix, '');
+      const prefixRegex = new RegExp(`^${prefix.replace(/\*/g, '[^/]+')}`);
```

Comment:
Constructing the regex this way would mean that user would be able to add random strings. Since we only support wildcard in place of segments of the URL, can you change it to something like this:

```suggestion
      const prefixRegex = new RegExp(
        `^(${prefix
          .split('.')
          .map((it) => (it === '*' ? '.+' : escape(it)))
          .join('\\.')})`
      );
```

We could do more complex stuff to support `*` anywhere while escaping the rest, but I think for now this is fine.

Reply:
I tested that, doesn't work. But work when I removed escape from that.

Reply:
What about `.+` Don't you think it should be `[^/]+`

Reply:
I think if we want to use escape this should be equal.
`escape(str) === str`
But they are not equal